Ad user story 25 - 26

adumortier opened the pull request:
- added the one-to-many relationship between `users` and `orders
- added a `status` column to the `Orders` table
- added the model tests of the relationships between `users` and `orders`
- made changes to the `create` action method of the `Order` controller to create the `order through the `current_user`
- added an `index` view page for `orders` and a route

adumortier committed:
Add a conditional in the cart show page to show a warning a message to visitors

adumortier committed:
Add test for relationship between users and orders

adumortier committed:
Add a relationship between user and orders and an enum to check the status of an order

adumortier committed:
Add the relationship between users and orders

adumortier committed:
Add migration to add a foreign key to orders

adumortier committed:
Add the status column to the Orders table

adumortier committed:
Apply the relationship between order and user and the addition to the Orders table

adumortier committed:
Add a test to allow a registered user to checkout and place an order

adumortier committed:
Add a route to view the current_user's orders

adumortier committed:
Add the index page that lists the orders of the current_user

adumortier committed:
Add creation of the order through the current_user

adumortier committed:
Merge branch 'master' of github.com:TravisBorgsmiller/monster_shop_1911 into ad_user_story_25

adumortier committed:
Fixed to check if there is registered user

KellyIB commented on app/models/order.rb:
```diff
@@ -4,6 +4,10 @@ class Order <ApplicationRecord
   has_many :item_orders
   has_many :items, through: :item_orders
 
+  belongs_to :user, optional: true
```
I am not sure we should have an optional qualifier here, shouldn't orders always belong to a user?  I think in the visitor's cart, they are not allowed to checkout and create an order unless they login.

cgaddis36 replied:
nice catch Kelly!

KellyIB reviewed (commented):
I think we might want to remove the optional qualifier in the order model.  I think we need to update any previous tests that fail with the new relationship we have here.  I don't think an order should be able to exist without a user to create it.

cgaddis36 commented on db/schema.rb:
```diff
@@ -57,6 +57,9 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.bigint "user_id"
+    t.integer "status", default: 0
```
Nice job on the relationship table, looks good

KellyIB committed:
Update order.rb

Removed optional:true qualifier.  I can tweak the tests when I pull this down, if that helps?

KellyIB reviewed (approved):
I think we might want to remove the optional qualifier in the order model.  I think we need to update any previous tests that fail with the new relationship we have here.  I don't think an order should be able to exist without a user to create it.

cgaddis36 commented on spec/features/cart/checkout_spec.rb:
```diff
@@ -27,7 +28,104 @@
+      expect(Order.last.zip).to eq(zip_code.to_i)
+      expect(Order.last.pending?).to eq(true)
+      expect(Order.last.user).to eq(User.last)
+
```
Nice specificity on testing the order attributes

cgaddis36 commented on spec/features/cart/checkout_spec.rb:
```diff
@@ -27,7 +28,104 @@
+# When I add items to my cart
+# And I visit my cart
+# I see a button or link indicating that I can check out
+# And I click the button or link to check out and fill out order info and click create order
```
Comments! :)

cgaddis36 commented on spec/models/user_spec.rb:
```diff
@@ -11,6 +11,11 @@
   end
+
+  describe "relationships" do 
+    it {should have_many :orders}
```
relationships setup looks good

cgaddis36 reviewed (approved):
Everything looks good, orders for cart implementation functional and ready for merge
